server/util.go

```diff
@@ -462,7 +462,8 @@ func DefaultBaseappOptions(appOpts types.AppOptions) []func(*baseapp.BaseApp) {
 	chainID := cast.ToString(appOpts.Get(flags.FlagChainID))
 	if chainID == "" {
 		// read the chainID from home directory (either from comet or genesis).
-		chainId, err := readChainIdFromHome(homeDir)
+		dbBackend := cast.ToString(appOpts.Get("db_backend"))
```

Comment:
Actually, we may want to use `GetAppDBBackend` instead

Reply:
we should parse the `db_backend` from `config.toml` specifically, it could be different from the app db backend.

Reply:
How about revert the original change, one can always set the chain-id specifically if the genesis file is not correct.

Reply:
GetAppDBBackend parses both

Reply:
The idea was to get a better UX, so imho we shouldn't revert the original PR.

Reply:
> GetAppDBBackend parses both

It shouldn't parse both, it should parse the cometbft one specifically, otherwise when cometbft backend is different from app one, it's still incorrect.

Reply:
Yes, I realized afterwards :D